handlers/challenge_handler.py

```diff
@@ -603,24 +603,27 @@ class SolveCommand(Command):
     @classmethod
     def execute(cls, slack_wrapper, args, timestamp, channel_id, user_id, user_is_admin):
         """Execute the Solve command."""
-        challenge = ""
-
-        if args:
-            challenge = get_challenge_from_args(ChallengeHandler.DB, args, channel_id)
 
-            if not challenge:
-                challenge = get_challenge_by_channel_id(ChallengeHandler.DB, channel_id)
+        # Get the challenge from the channel id or the first argument to !solve.
+        challenge = get_challenge_from_args(ChallengeHandler.DB, args, channel_id)
+        if challenge:
+            if channel_id == challenge.channel_id:
+                # !solve called from the challenge - additional args are
+                # assumed to be other solvers.
                 additional_args = args if args else []
-            else:
+            elif channel_id == challenge.ctf_channel_id:
+                # !solve called from the ctf - first argument must be the name
+                # of the challenge.
                 additional_args = args[1:] if len(args) > 1 else []
+            else:
+                # Prevent solves from happening outside the ctf/challenge
+                # rooms.
+                # NOTE: This branch should be unreachable.
+                raise InvalidCommand("You must be in a CTF or Challenge channel to use this command.")
         else:
-            # No arguments => direct way of resolving challenge
-            challenge = get_challenge_by_channel_id(ChallengeHandler.DB, channel_id)
-
-            additional_args = []
-
-        if not challenge:
-            raise InvalidCommand("This challenge does not exist.")
+            # The user made a typo, or is perhaps confused.
+            # NOTE: This branch should be unreachable.
+            raise InvalidCommand("You must be in a CTF or Challenge channel to use this command.")
```

Comment:
move this to the top, so `if not challenge: raise Invalid` and then untab everything

Reply:
This is done, resolving.